src/HooglePlus/FilterTest.hs

```diff
@@ -168,4 +171,30 @@ handleAnyException = (`catch` (return . handler . show :: SomeException -> IO (M
   where
     handler ex  | "timeout" `isInfixOf` ex = Nothing
                 | otherwise = (Just . Left. UnknownError) ex
-fmtFunction_ = printf "((%s) :: %s) %s" :: String -> String -> String -> String
\ No newline at end of file
+fmtFunction_ = printf "((%s) :: %s) %s" :: String -> String -> String -> String
+
+checkDuplicates :: MonadIO m => [String] -> String -> String -> FilterTest m Bool
+checkDuplicates modules sigStr body = do
+  st <- get
+  case sampleResults st of
+    Nothing -> do
+      result' <- liftIO (Just . (`SampleResult` []) <$> generateInputs modules funcSig)
```

Comment:
I'm confused about what the `result'` represent. It _seems_ like this fills random inputs into the `_results` field of the `sampleResult` type. But when does the `_input` field get filled in?

Reply:
`result'` is a bad name and refers to the updated `result :: SampleResult` (which got removed in this commit). Thanks for catching this

Reply:
The function would generate multiple inputs and fill in the `_input` field, and then evaluate our inputs with the updated `SampleResult`.